File: 392.Is_Subsequence/main.go
package main

import "fmt"

type Solution struct{}

func (s Solution) isSubsequence(sub string, t string) bool {
    sIndex, tIndex := 0, 0
    for sIndex < len(sub) && tIndex < len(t) {
        if sub[sIndex] == t[tIndex] {
            sIndex++
        }
        tIndex++
    }
    return sIndex == len(sub)
}

func main() {
    solution := Solution{}
    s := "abc"
    t := "ahbgdc"
    fmt.Println(solution.isSubsequence(s, t)) // Output: true
}

File: 392.Is_Subsequence/main.java
public class Solution {
    public boolean isSubsequence(String s, String t) {
        int sIndex = 0, tIndex = 0;
        while (sIndex < s.length() && tIndex < t.length()) {
            if (s.charAt(sIndex) == t.charAt(tIndex)) {
                sIndex++;
            }
            tIndex++;
        }
        return sIndex == s.length();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "abc";
        String t = "ahbgdc";
        System.out.println(solution.isSubsequence(s, t)); // Output: true
    }
}
